src/data/get_runway_stats.py

```diff
@@ -21,14 +21,23 @@ class RunwayStats:
     _runways_stats_df: pd.DataFrame = field(init=False)
 
     def __post_init__(self, runways: pd.DataFrame) -> None:
-        self._runways_stats_df = runways.map(self.count_runways)
-        self._runways_stats_df['total_runway_length'] = runways.map(self.sum_runways_len)
-
-    def count_runways(self, airport_runways: list[RunwayDict]) -> int | None:
-        return len(airport_runways) or None
-
-    def sum_runways_len(self, airport_runways: list[RunwayDict]) -> int | None:
-        return sum(runway['length_in_ft'] for runway in airport_runways) or None
+        runways_count = list(map(self.count_runways, runways['runways'], runways['iata']))
+        runways_length = list(map(self.sum_runways_len, runways['runways'],runways['iata']))
+        runways_count_df = pd.DataFrame(runways_count, columns = ['iata', 'runways'])
+        runways_length_df = pd.DataFrame(runways_length, columns = ['iata', 'total_runway_length'])
+        self._runways_stats_df = pd.merge(runways_count_df, runways_length_df, on = 'iata')
```

Comment:
This seems far too complex to me. I think the same could be achieved by constructing a list of dictionaries in place that then gets converted to a dataframe at the end? Or something similar perhaps

Reply:
Okay yeah I'll try that